File: applications/luci-app-statistics/luasrc/model/cbi/luci_statistics/ping.lua
--[[

Luci configuration model for statistics - collectd ping plugin configuration
(c) 2008 Freifunk Leipzig / Jo-Philipp Wich <xm@leipzig.freifunk.net>

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

$Id$

]]--

m = Map("luci_statistics",
	translate("Ping Plugin Configuration"),
	translate(
		"The ping plugin will send icmp echo replies to selected " ..
		"hosts and measure the roundtrip time for each host."
	))

-- collectd_ping config section
s = m:section( NamedSection, "collectd_ping", "luci_statistics" )

-- collectd_ping.enable
enable = s:option( Flag, "enable", translate("Enable this plugin") )
enable.default = 0

-- collectd_ping.hosts (Host)
hosts = s:option( Value, "Hosts", translate("Monitor hosts"), translate ("Add multiple hosts separated by space."))
hosts.default = "127.0.0.1"
hosts:depends( "enable", 1 )

-- collectd_ping.ttl (TTL)
ttl = s:option( Value, "TTL", translate("TTL for ping packets") )
ttl.isinteger = true
ttl.default   = 128
ttl:depends( "enable", 1 )

-- collectd_ping.interval (Interval)
interval = s:option( Value, "Interval", translate("Interval for pings"), translate ("Seconds") )
interval.isinteger = true
interval.default   = 30
interval:depends( "enable", 1 )

return m

File: applications/luci-app-statistics/htdocs/luci-static/resources/view/statistics/plugins/ping.js
'use strict';
'require baseclass';
'require form';

return baseclass.extend({
	title: _('Ping Plugin Configuration'),
	description: _('The ping plugin will send icmp echo replies to selected hosts and measure the roundtrip time for each host.'),

	addFormOptions: function(s) {
		var o;

		o = s.option(form.Flag, 'enable', _('Enable this plugin'));

		o = s.option(form.DynamicList, 'Hosts', _('Monitor hosts'));
		o.default = '127.0.0.1';
		o.datatype = 'host';
		o.depends('enable', '1');

		o = s.option(form.ListValue, 'AddressFamily', _('Address family'));
		o.default = 'any';
		o.depends('enable', '1');
		o.value('any');
		o.value('ipv4');
		o.value('ipv6');

		o = s.option(form.Value, 'TTL', _('TTL for ping packets'));
		o.default = '127';
		o.datatype = 'range(0, 255)';
		o.depends('enable', '1');

		o = s.option(form.Value, 'Interval', _('Interval for pings'), _('Seconds'));
		o.default = '30';
		o.datatype = 'ufloat';
		o.depends('enable', '1');

	    o=s.option(form.Value,'MaxMissed',_('Maximum Missed Packets'),
		       _('When a host has not replied to this number of packets in a row, re-resolve the hostname in DNS.  Useful for dynamic DNS hosts.  Default is -1 = disabled.'));
		o.placeholder = '-1';
		o.datatype = 'and(min(-1),integer)'
		o.optional = true;
		o.depends('enable', '1');
	},

	configSummary: function(section) {
		var hosts = L.toArray(section.Hosts);

		if (hosts.length)
			return N_(hosts.length, 'Monitoring one host', 'Monitoring %d hosts').format(hosts.length);
	}
});
